<Type Name="ForeignKeyReferenceAlreadyHasValueException" FullName="System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException">
  <Metadata>
    <Meta Name="ms.openlocfilehash" Value="d8770cffaeaf250a5417fb744564f86f9759ee9f" />
    <Meta Name="ms.sourcegitcommit" Value="6a0b904069161bbaec4ffd02aa7d9cf38c61e72e" />
    <Meta Name="ms.translationtype" Value="HT" />
    <Meta Name="ms.contentlocale" Value="fr-FR" />
    <Meta Name="ms.lasthandoff" Value="06/24/2018" />
    <Meta Name="ms.locfileid" Value="36689299" />
  </Metadata>
  <TypeSignature Language="C#" Value="public class ForeignKeyReferenceAlreadyHasValueException : InvalidOperationException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ForeignKeyReferenceAlreadyHasValueException extends System.InvalidOperationException" />
  <TypeSignature Language="DocId" Value="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" />
  <TypeSignature Language="VB.NET" Value="Public Class ForeignKeyReferenceAlreadyHasValueException&#xA;Inherits InvalidOperationException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ForeignKeyReferenceAlreadyHasValueException : InvalidOperationException" />
  <TypeSignature Language="F#" Value="type ForeignKeyReferenceAlreadyHasValueException = class&#xA;    inherit InvalidOperationException" />
  <AssemblyInfo>
    <AssemblyName>System.Data.Linq</AssemblyName>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.InvalidOperationException</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary>
      <span data-ttu-id="02380-101">Représente les erreurs qui se produisent lors d’une tentative de modification d’une clé étrangère alors que l’entité est déjà chargée.</span>
      <span class="sxs-lookup">
        <span data-stu-id="02380-101">Represents errors that occur when an attempt is made to change a foreign key when the entity is already loaded.</span>
      </span>
    </summary>
    <remarks>To be added.</remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary>
          <span data-ttu-id="02380-102">Initialise une nouvelle instance de la classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" />.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-102">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class.</span>
          </span>
        </summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters />
      <Docs>
        <summary>
          <span data-ttu-id="02380-103">Initialise une nouvelle instance de la classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> avec un message système qui décrit l'erreur.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-103">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a system-supplied message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="02380-104">Ce constructeur initialise la propriété <xref:System.Exception.Message%2A> de la nouvelle instance sur un message système qui décrit l'erreur, par exemple « DefaultMessageDisplayedByParameterlessCtorWriterMustSupply ».</span><span class="sxs-lookup"><span data-stu-id="02380-104">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "DefaultMessageDisplayedByParameterlessCtorWriterMustSupply".</span></span> <span data-ttu-id="02380-105">Ce message prend en compte la culture système en cours.</span><span class="sxs-lookup"><span data-stu-id="02380-105">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="02380-106">Le tableau suivant montre les valeurs initiales des propriétés d'une instance de <xref:System.InsufficientMemoryException>.</span><span class="sxs-lookup"><span data-stu-id="02380-106">The following table shows the initial property values for an instance of <xref:System.InsufficientMemoryException>.</span></span>  
  
|<span data-ttu-id="02380-107">Property</span><span class="sxs-lookup"><span data-stu-id="02380-107">Property</span></span>|<span data-ttu-id="02380-108">Value</span><span class="sxs-lookup"><span data-stu-id="02380-108">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="02380-109">`null`.</span><span class="sxs-lookup"><span data-stu-id="02380-109">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="02380-110">Chaîne du message d'erreur localisé.</span><span class="sxs-lookup"><span data-stu-id="02380-110">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException : string -&gt; System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" Usage="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="02380-111">Message décrivant l'exception.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-111">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="02380-112">L'appelant de ce constructeur doit vérifier que cette chaîne a été localisée pour la culture en cours du système.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-112">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="02380-113">Initialise une nouvelle instance de la classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> avec un message spécifié décrivant l'erreur.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-113">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a specified message that describes the error.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="02380-114">Le tableau suivant montre les valeurs initiales des propriétés d'une instance de <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span><span class="sxs-lookup"><span data-stu-id="02380-114">The following table shows the initial property values for an instance of <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span></span>  
  
|<span data-ttu-id="02380-115">Property</span><span class="sxs-lookup"><span data-stu-id="02380-115">Property</span></span>|<span data-ttu-id="02380-116">Value</span><span class="sxs-lookup"><span data-stu-id="02380-116">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="02380-117">`null`.</span><span class="sxs-lookup"><span data-stu-id="02380-117">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="02380-118">Chaîne de message d'erreur spécifiée dans `message`.</span><span class="sxs-lookup"><span data-stu-id="02380-118">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException : string * Exception -&gt; System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" Usage="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message">
          <span data-ttu-id="02380-119">Message décrivant l'exception.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-119">The message that describes the exception.</span>
          </span>
          <span data-ttu-id="02380-120">L'appelant de ce constructeur doit vérifier que cette chaîne a été localisée pour la culture en cours du système.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-120">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span>
          </span>
        </param>
        <param name="innerException">
          <span data-ttu-id="02380-121">Exception ayant provoqué l'exception actuelle.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-121">The exception that is the cause of the current exception.</span>
          </span>
          <span data-ttu-id="02380-122">Si le paramètre <c>innerException</c> n’est pas <see langword="null" />, l’exception actuelle est levée dans un bloc <see langword="catch" /> qui gère l’exception interne.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-122">If the <c>innerException</c> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span>
          </span>
        </param>
        <summary>
          <span data-ttu-id="02380-123">Initialise une nouvelle instance de la classe <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> avec un message d'erreur spécifié et une référence à l'exception interne ayant provoqué cette exception.</span>
          <span class="sxs-lookup">
            <span data-stu-id="02380-123">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span>
          </span>
        </summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="02380-124">Une exception levée en conséquence directe d'une exception précédente doit contenir une référence à celle-ci dans la propriété <xref:System.Exception.InnerException%2A>.</span><span class="sxs-lookup"><span data-stu-id="02380-124">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="02380-125">La propriété <xref:System.Exception.InnerException%2A> retourne la même valeur que celle qui a été passée dans le constructeur ou retourne `null` si la propriété <xref:System.Exception.InnerException%2A> ne fournit pas la valeur d'exception interne au constructeur.</span><span class="sxs-lookup"><span data-stu-id="02380-125">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="02380-126">Le tableau suivant montre les valeurs initiales des propriétés d'une instance de <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span><span class="sxs-lookup"><span data-stu-id="02380-126">The following table shows the initial property values for an instance of <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span></span>  
  
|<span data-ttu-id="02380-127">Property</span><span class="sxs-lookup"><span data-stu-id="02380-127">Property</span></span>|<span data-ttu-id="02380-128">Value</span><span class="sxs-lookup"><span data-stu-id="02380-128">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="02380-129">`null`.</span><span class="sxs-lookup"><span data-stu-id="02380-129">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="02380-130">Chaîne de message d'erreur spécifiée dans `message`.</span><span class="sxs-lookup"><span data-stu-id="02380-130">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
      </Docs>
    </Member>
  </Members>
</Type>